easel/lib/widgets/image_widget.dart

```diff
@@ -11,31 +13,47 @@ class ImageWidget extends StatelessWidget {
 
   const ImageWidget({Key? key, this.file, this.filePath}) : super(key: key);
 
-  @override
-  Widget build(BuildContext context) {
-    return file != null
-        ? Image.memory(
-            file!.readAsBytesSync(),
-            width: 1.sw,
-            height: 1.sh,
-            fit: BoxFit.fitHeight,
-          )
-        : SizedBox(
-            height: double.infinity,
-            child: CachedNetworkImage(
-              fit: BoxFit.fitHeight,
-              imageUrl: filePath!,
-              errorWidget: (a, b, c) =>
-                  const Center(child: Icon(Icons.error_outline)),
-              placeholder: (context, url) => Center(
-                child: SizedBox(
-                  height: 50.0.h,
-                  child: Image.asset(
-                    kLoadingGif,
-                  ),
-                ),
+  EaselProvider get easelProvider => sl();
+
+  Widget buildImage() {
+    if (file == null) {
+      return SizedBox(
+        height: double.infinity,
+        child: CachedNetworkImage(
+          fit: BoxFit.fitHeight,
+          imageUrl: filePath!,
+          errorWidget: (a, b, c) => const Center(child: Icon(Icons.error_outline)),
+          placeholder: (context, url) => Center(
+            child: SizedBox(
+              height: 50.0.h,
+              child: Image.asset(
+                kLoadingGif,
               ),
             ),
-          );
+          ),
+        ),
+      );
+    }
+
+    return Image.memory(
+      file!.readAsBytesSync(),
+      width: 1.sw,
+      height: 1.sh,
+      fit: easelProvider.imageCheckBox ? BoxFit.fitHeight : BoxFit.fitWidth,
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return WillPopScope(
+        child: Stack(
```

Comment:
if we remove stack we dont have to give height and width.

Reply:
Removed the stack, It was for the checkbox but checkbox was shifted to preview_screen but the stack was still here.